### PR TITLE
Add release channels list support

### PR DESCRIPTION
**Pull request by sfc-gh-melnacouzi**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [ ] I've added or updated integration tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually on MacOS.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually on Windows.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [x] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
Add support for `snow app release-channel list [CHANNEL]` command.

Integ tests will be added once UIParam is in production

Screenshot 2024-12-16 at 11 36 30 AM

Screenshot 2024-12-16 at 11 40 49 AM

### CODE REVIEW

File: src/snowflake/cli/_plugins/nativeapp/entities/application_package.py [hunk 1]
```diff
@@ -134,6 +136,99 @@ def ensure_app_roles_is_a_set(
         return application_roles
 
 
+class ReleaseChannel(dict[str, Any]):
```
Comment (sfc-gh-gbloom) [inline src/snowflake/cli/_plugins/nativeapp/entities/application_package.py:139]: Maybe specify the exact dict structure we're expecting with a `TypedDict`?
It's the same type that `show_release_channels()` returns, right? We can reuse it there too.
That way we would also be able to remove the `type: ignore` comments below.

File: src/snowflake/cli/_plugins/nativeapp/entities/application_package.py [hunk 2]
```diff
@@ -134,6 +136,99 @@ def ensure_app_roles_is_a_set(
...
+
+    @property
+    def targets(self) -> dict[str, Any]:
+        return self.get("targets")  # type: ignore
+
+    @property
+    def versions(self) -> list[str]:
+        return self.get("versions")  # type: ignore
+
+    def print_to_console(self, console: AbstractConsole) -> None:
```
Comment (sfc-gh-gbloom) [inline src/snowflake/cli/_plugins/nativeapp/entities/application_package.py:191]: Can you add a screenshot with an output example?

File: src/snowflake/cli/_plugins/nativeapp/entities/application_package.py [hunk 3]
```diff
@@ -134,6 +136,99 @@ def ensure_app_roles_is_a_set(
...
+        console.message(f"""[bold]{self.name}[/bold]""")
+        accounts_list: Optional[list[str]] = self.targets.get("accounts")
+        target_accounts = (
+            f"({', '.join(accounts_list)})"
+            if accounts_list is not None
+            else "ALL ACCOUNTS"
+        )
+
+        formatted_created_on = (
+            self.created_on.astimezone().strftime("%Y-%m-%d %H:%M:%S.%f %Z")
```
Comment (sfc-gh-gbloom) [inline src/snowflake/cli/_plugins/nativeapp/entities/application_package.py:204]: nit:
Why this specific format? Do we already print some datetimes elsewhere?

Comment (sfc-gh-melnacouzi) [reply src/snowflake/cli/_plugins/nativeapp/entities/application_package.py:204]: From the design doc, Bruno wanted something more human readable for console output - in this case, we are outputting the current timezone. 
Also, this is not set in stone, as it's only a console output. `--format=json` part is printed using the default.

**Review by sfc-gh-gbloom — APPROVED** [review]

**Pull request merged**